stackslib/src/chainstate/stacks/boot/pox-4-vote.clar

```diff
@@ -0,0 +1,56 @@
+;;
```

Comment:
A couple of overall things about this contract:

* There doesn't need to be a `last-round` data var.  The signers themselves decide what the round number is.  For now, the node would accept the first aggregate public key that receives 70% of the votes in a given reward cycle.  In the future, the round would be used to allow DKG to happen mid-cycle, but it's currently unknown if that will ship with Nakamoto or sBTC.

* The node will track aggregate public key votes via transaction event receipts, which it will inspect in the chains coordinator.  Can you add `(print )` statements for when a vote happens?  Then, the node can just inspect the `votes` and `tally` maps at the end of the block.

Reply:
Tracking the last-round does actually help. 

Stacks-signers do not currently track a round. They track rounds internally in wsts but I could see it being useful to have this globally stored somewhere.

Suppose Alice wants to do a DKG round but she just came online after Bob triggered DKG round 1. She did not witness this round complete. She will need to know what the last round number used to vote was so she can trigger a new DKG round with last_round + 1. This will result in her triggering a DKG round for an outdated number that will subsequently get ignored by the contract as it will simply see signers attempting to resubmit votes when they have already voted.

Reply:
last round map has been added